### ISSUE TITLE
Type error with usePermission declaration

### ISSUE THREAD

**Issue by danielroe** [antfu/vueuse#31]
When using the current version of this package and TypeScript 3.7.4, I get the following error on compilation:

```
ERROR in /my/folder/node_modules/@vueuse/core/index.d.ts(201,138):
201:138 Cannot find name 'State'.
    199 |     name: DescriptorNamePolyfill;
    200 | };
  > 201 | declare function usePermission(permissionDesc: GeneralPermissionDescriptor | PermissionDescriptor['name'] | DescriptorNamePolyfill): Ref<State>;
        |                                                                                                                                          ^
    202 |
    203 | declare function usePreferredColorScheme(): Readonly<Ref<"dark" | "light" | "no-preference">>;
    204 |
```

This appears to be because the function signature is exported but not the type from `usePermission`:

https://github.com/antfu/vueuse/blob/91a5c40130d195fdef42a36d9b3f5bbf12eff190/packages/core/usePermission/index.ts#L13

**Comment by antfu** [comment]
@danielroe I thought this should be handled correctly by Typescript 🤔. FIxed in v0.0.16. Thanks!

**Comment by danielroe** [comment]
@antfu I agree, it seems TypeScript being idiosyncratic! Thanks for the great library 😀
